x-pack/platform/packages/shared/onechat/onechat-server/src/tools.ts

```diff
@@ -45,6 +45,81 @@ export interface RegisteredTool<
   meta?: RegisteredToolMeta;
 }
 
+/**
+ * Onechat tool, as registered by built-in tool providers.
+ */
+export interface EsqlToolDefinition extends ToolDescriptor {
+  /**
+   * The ESQL Tool name.
+   */
+  name: string;
+
+  /**
+   * The ESQL query to be executed.
+   */
+  query: string;
+
+  /**
+   * Parameters that can be used in the query.
+   * Each parameter has a key identifier and metadata about its type and usage.
+   */
+  params: Record<
+    string,
+    {
+      /**
+       * The data type of the parameter.
+       */
+      type: string;
```

Comment:
i suggest this to be an enum of value types. To start with we can define this to be just string and number.